resources/services/backup/vaults.go

```diff
@@ -0,0 +1,389 @@
+					},
+				},
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+
+func fetchBackupVaults(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	cl := meta.(*client.Client)
+	svc := cl.Services().Backup
+	params := backup.ListBackupVaultsInput{MaxResults: aws.Int32(100)}
+	for {
+		result, err := svc.ListBackupVaults(ctx, &params, func(o *backup.Options) {
+			o.Region = cl.Region
+		})
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		res <- result.BackupVaultList
+		if aws.ToString(result.NextToken) == "" {
+			break
+		}
+		params.NextToken = result.NextToken
+	}
+	return nil
+}
+
+func resolveVaultTags(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	vault := resource.Item.(types.BackupVaultListMember)
+	cl := meta.(*client.Client)
+	svc := cl.Services().Backup
+	params := backup.ListTagsInput{ResourceArn: vault.BackupVaultArn}
+	tags := make(map[string]string)
+	for {
+		result, err := svc.ListTags(ctx, &params, func(o *backup.Options) { o.Region = cl.Region })
+		if result == nil {
+			break
+		}
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		for k, v := range result.Tags {
+			tags[k] = v
+		}
+		if aws.ToString(result.NextToken) == "" {
+			break
+		}
+		params.NextToken = result.NextToken
+	}
+	return diag.WrapError(resource.Set(c.Name, tags))
+}
+
+func resolveVaultAccessPolicy(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	vault := resource.Item.(types.BackupVaultListMember)
+	cl := meta.(*client.Client)
+	svc := cl.Services().Backup
+	result, err := svc.GetBackupVaultAccessPolicy(
+		ctx,
+		&backup.GetBackupVaultAccessPolicyInput{BackupVaultName: vault.BackupVaultName},
+		func(o *backup.Options) { o.Region = cl.Region },
+	)
+	if err != nil {
+		if cl.IsNotFoundError(err) {
+			return nil
+		}
+		return diag.WrapError(err)
+	}
+	return diag.WrapError(resource.Set(c.Name, result.Policy))
+}
+
+func resolveVaultNotifications(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource) error {
+	vault := resource.Item.(types.BackupVaultListMember)
+	cl := meta.(*client.Client)
+	svc := cl.Services().Backup
+	result, err := svc.GetBackupVaultNotifications(
+		ctx,
+		&backup.GetBackupVaultNotificationsInput{BackupVaultName: vault.BackupVaultName},
+		func(o *backup.Options) { o.Region = cl.Region },
+	)
+	if err != nil {
+		var ae smithy.APIError
+		if !errors.As(err, &ae) {
+			return diag.WrapError(err)
+		}
+		if ae.ErrorCode() == "ERROR_2106" {
+			// trying to ignore "ERROR_2106: Failed reading notifications from database for Backup vault ..."
+			return nil
+		}
```

Comment:
What is this?

Reply:
if there are no notifications configured for a vault this API request returns that error so it can't be matched with `client.IsNotFoundError`

Reply:
If there is no notifications attached to a vault then we have this weird error from the API which we cannot match with the already existing `client.IsNotFoundError`.